.github/workflows/unit-testing-and-coverage.yml

```diff
@@ -56,7 +56,11 @@ jobs:
         # language=bash
         run: |
           pip install pytest pytest-mock check-manifest
-          git fetch --no-tags --prune --no-recurse-submodules --depth=1 origin master:master
+          branch_name="$(git symbolic-ref HEAD 2> /dev/null)"
+          # Fetch master if needed because some test cases need its existence
+          if [[ ${branch_name##refs/heads/} != master ]]; then
+            git fetch --no-tags --prune --no-recurse-submodules --depth=1 origin master:master
+          fi
```

Comment:
Selective fetching: Do not fetch if we're already inside master.

This should prevent GA checks erroring out if changes are pushed directly to `master`